src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -611,7 +622,14 @@ private[fastq] object FastqDemultiplexer {
     *                      for the unmatched sample.
     * @param records the records, one for each read that has template bases.
     */
-  case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord])
+  case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord], passQc: Boolean = true) {
+
+    def keep(keepOnlyPassing: Boolean = false): Boolean = {
```

Comment:
Add a doc-string :)
```suggestion
    /** Returns true if this [[DemuxResult]] should be kept for output, false otherwise.
     *
     * Returns true if `keepOnlyPassing` is false or if `passQc` is true
     *
     * @param keepOnlyPassing true to keep only passing reads, false to keep all reads
    */
    def keep(keepOnlyPassing: Boolean = false): Boolean = {
```